### PR TITLE
TRUNK-6145: Allow validation to be disabled only within the current t…

### PR DESCRIPTION
**Pull request by mseaton**
…hread

### CODE REVIEW

File: api/src/main/java/org/openmrs/validator/ValidateUtil.java [hunk 1]
```diff
@@ -146,4 +149,31 @@ public static void setDisableValidation(Boolean disableValidation) {
...
+	 * @return true if validation has been disabled for the current thread, false otherwise
+	 * @since 2.5.8
+	 */
+	public static boolean isValidationDisabledForThread() {
+		return disableValidationForThread.get() == Boolean.TRUE;
+	}
+
+	/**
+	 * Sets a ThreadLocal variable to indicate that validation should be disabled for the current thread
+	 * NOTE: This should always be used in conjunction with the resumeValidationForThread method to
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/validator/ValidateUtil.java:162]: Just a minor comment, i would not include the implementation details `Sets a ThreadLocal variable` in the method description.

File: api/src/main/java/org/openmrs/validator/ValidateUtil.java [hunk 2]
```diff
@@ -146,4 +149,31 @@ public static void setDisableValidation(Boolean disableValidation) {
...
+	 * NOTE: This should always be used in conjunction with the resumeValidationForThread method to 
+	 * ensure the ThreadLocal instance is cleaned up at the end of the operation
+	 * @since 2.5.8
+	 */
+	public static void disableValidationForThread() {
+		disableValidationForThread.set(Boolean.TRUE);
+	}
+
+	/**
+	 * Removes a ThreadLocal variable to indicate that validation should be re-enabled for the current thread
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/validator/ValidateUtil.java:171]: Same as above.

Comment (mseaton) [reply api/src/main/java/org/openmrs/validator/ValidateUtil.java]: Changed

File: api/src/main/java/org/openmrs/validator/ValidateUtil.java (hunk 1 above)
Comment (mseaton) [reply api/src/main/java/org/openmrs/validator/ValidateUtil.java]: Changed

**Pull request merged**